lib/node/http.js

```diff
@@ -383,6 +385,108 @@ class HTTP extends Server {
 
       res.json(200, { success: true });
     });
+
+    this.get('/info/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid parameter.');
+
+      const network = this.network;
+      const height = this.chain.height;
+      const nameHash = rules.hashName(name);
+      const reserved = rules.isReserved(nameHash, height + 1, network);
+      const [start, week] = rules.getRollout(nameHash, network);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      let info = null;
+
+      if (ns) {
+        if (!ns.isExpired(height, network))
+          info = ns.getJSON(height, network);
+      }
+
+      return res.json(200, {
+        start: {
+          reserved: reserved,
+          week: week,
+          start: start
+        },
+        info
+      });
+    });
+
+    this.get('/name/hash/:hash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const hash = valid.bhash('hash');
+
+      if (!hash)
+        throw new Error('Invalid hash.');
+
+      const ns = await this.chain.db.getNameState(hash);
+
+      if (!ns)
+        return res.json(404);
+
+      return res.json(200, { name: ns.name.toString('binary') });
```

Comment:
`ns.getJSON` should be the 2nd argument to `res.json` like this:

See implementation here:
https://github.com/handshake-org/hsd/blob/696dfc4601ef5e8cd954cde727142f39c965fcf6/lib/wallet/rpc.js#L1774

https://github.com/handshake-org/hsd/blob/master/lib/covenants/namestate.js#L651